ot2/octopus-full-ssh.py

```diff
@@ -0,0 +1,590 @@
+        self.pre_pool_plate_2 = self.protocol.load_labware('appliedbiosystems_384_wellplate_40ul', '2', label='Pre-pool Plate 2 (5-8)')
+
+        SSH_MODE = self.ssh_mode
+        if SSH_MODE:
+            self.pool_plate.set_offset(x=0.10, y=1.60, z=0.00)
+            self.dna_plate.set_offset(x=0.40, y=1.10, z=0.10)
+            self.consolidation_plate.set_offset(x=0.30, y=1.10, z=0.00)
+            self.pre_pool_plate_1.set_offset(x=0.00, y=0.80, z=0.60)
+            self.pre_pool_plate_2.set_offset(x=0.00, y=0.80, z=0.60)
+
+        ### reagent locations
+        DNA_pool = self.pool_plate.wells_by_name()['A1']
+        Purification_Beads = self.pool_plate.wells_by_name()['A3']
+        Purification_Beads_discard = self.pool_plate.wells_by_name()['A4']
+        Ethanol = [self.pool_plate.wells_by_name()['A6'], self.pool_plate.wells_by_name()['A9']]
+        Ethanol_discard = [self.pool_plate.wells_by_name()['A7'], self.pool_plate.wells_by_name()['A10']]
+        Tris_HCl = self.pool_plate.wells_by_name()['A12']
+
+        ### consolidate Reaction A plate after iGenomX_1 into consolidation plate
+        self.set_flow_rate(48, 48, 48)
+            # consolidating VE plates 1-4 to Pool Plate columns 1-4
+        if NUM_PLATES >= 1:
+            self.p300.consolidate(5, [self.pre_pool_plate_1.rows_by_name()['A'][num_plate].bottom(z=0.5) for num_plate in range(0, 23, 2)], self.consolidation_plate.columns('1'))
+        if NUM_PLATES >= 2:
+            self.p300.consolidate(5, [self.pre_pool_plate_1.rows_by_name()['A'][num_plate].bottom(z=0.5) for num_plate in range(1, 24, 2)], self.consolidation_plate.columns('2'))
+        if NUM_PLATES >= 3:
+            self.p300.consolidate(5, [self.pre_pool_plate_1.rows_by_name()['B'][num_plate].bottom(z=0.5) for num_plate in range(0, 23, 2)], self.consolidation_plate.columns('3'))
+        if NUM_PLATES >= 4:
+            self.p300.consolidate(5, [self.pre_pool_plate_1.rows_by_name()['B'][num_plate].bottom(z=0.5) for num_plate in range(1, 24, 2)], self.consolidation_plate.columns('4'))
+            # consolidating VE plates 5-8 to Pool Plate columns 5-8
+        if NUM_PLATES >= 5:
+            self.p300.consolidate(5, [self.pre_pool_plate_2.rows_by_name()['A'][num_plate].bottom(z=0.5) for num_plate in range(0, 23, 2)], self.consolidation_plate.columns('5'))
+        if NUM_PLATES >= 6:
+            self.p300.consolidate(5, [self.pre_pool_plate_2.rows_by_name()['A'][num_plate].bottom(z=0.5) for num_plate in range(1, 24, 2)], self.consolidation_plate.columns('6'))
+        if NUM_PLATES >= 7:
+            self.p300.consolidate(5, [self.pre_pool_plate_2.rows_by_name()['B'][num_plate].bottom(z=0.5) for num_plate in range(0, 23, 2)], self.consolidation_plate.columns('7'))
+        if NUM_PLATES == 8:
+            self.p300.consolidate(5, [self.pre_pool_plate_2.rows_by_name()['B'][num_plate].bottom(z=0.5) for num_plate in range(1, 24, 2)], self.consolidation_plate.columns('8'))
```

Comment:
I'm getting a bad smell here. I realize it might be more explicit just to write all these commands out, but is there a reason why this wouldn't work?

```py3
for plate_num in range(1, NUM_PLATES + 1):
    pre_pool_plate = self.pre_pool_plate_1 if plate_num > 4 else self.pre_pool_plate2
    if plate_num % 2 == 1:
        wells = [self.pre_pool_plate1.rows_by_name()['B'][row_n].bottom(z=0.5) for row_n in range(0, 23, 2)]
    else:
        wells = [self.pre_pool_plate1.rows_by_name()['A'][row_n].bottom(z=0.5) for row_n in range(1, 24, 2)]
    
    self.p300.consolidate(volume=5, source=wells, dest=self.consolidation_plate.columns(str(plate_num))

Reply:
Yeah... I was going back and forth on how that part should be written but ended up keeping it more explicit. The reason was for user flexibility. 

For example, there might be a case where the runner only has 2 separate pre-stocked 384-well reagent plates `pre_pool_plate_1` `pre_pool_plate_2` prepped for, say, 3-plate and 2-plate runs respectively. When a 5-plate run is required, the runner could use those two pre-stocked plates and then just edit the protocol to reflect which columns to consolidate from.

The protocol by default will assume `pre_pool_plate_1` to contain 4 plates and `pre_pool_plate_2` to contain 1 plate. Having the code more explicit makes it arguably more straightforward to adapt the protocol to such a scenario as to avoid having to take the time pre-stocking reagent plates _fresh_.

Having something like what you proposed would definitely make this block more neat and tidy. However, the rows and columns alternate in separate patterns (not just singly even-odd for both). That is, the rows `rows_by_name()['A']` and `rows_by_name()['B']` alternate every two consolidation steps while the columns in `range(0, 23, 2)` and `range(1, 24, 2)` alternate every single consolidation step. I just felt that generalizing this consolidation pattern for the application where the maximum consolidation steps is only `8` would be unnecessarily restricting.